merkle.go

```diff
@@ -7,14 +7,22 @@ import (
 	"sort"
 )
 
-var ErrorIncompleteTree = errors.New("number of leaves must be a power of 2")
+// PaddingValue is used for padding unbalanced trees. This value should not be permitted at the leaf layer to
+// distinguish padding from actual members of the tree.
+var PaddingValue = node{
+	value: []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
```

Comment:
I've changed this line to:
```golang
value: make([]byte, NodeSize), // Zero filled.
```

I initially chose the original since I felt it more explicitly sets value, but your comment changed my mind about what's better.